### PR TITLE
App Submission: DCRDEX

### PR DESCRIPTION
**Pull request by peterzen**
# App Submission: DCRDEX

### DCRDEX
...

### 256x256 SVG icon
https://svgshare.com/i/pNb.svg
...

### Gallery images
  - https://i.imgur.com/dkqCPtJ.png
  - https://i.imgur.com/Mc3aTnR.png
  - https://i.imgur.com/TFFU0ur.png
...

### I have tested my app on:
- [X] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [X] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [X] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

### CODE REVIEW

**Comment by nmfretz** [comment]
Sorry for the delay on this @peterzen. Thanks very much for submitting the DCRDEX app. Currently, for crypto-related apps, we're exclusively accepting apps that solely focus on bitcoin for the official Umbrel App Store.

That being said, you can still create a Community App Store to distribute DCRDEX to Umbrel users:
https://github.com/getumbrel/umbrel-community-app-store

I'd be happy to help with this process and can still review the code you have submitted here to make sure DCRDEX runs smoothly on Umbrel.

**Comment by peterzen** [comment]
Thank you @nmfretz for looking into this app submission.  We are currently working on the Community App Store, if you could please review the code that would be much appreciated.

https://github.com/peterzen/umbrel-decred-app-store

Many thanks!

**Comment by chappjc** [comment]
It might be worth noting that this application provides a neutrino-powered (privacy preserving) Bitcoin wallet, no other dependencies required.  It's not a lie to say that despite the name DCRDEX, the application prioritizes it's native BTC wallet.

File: decred-dcrdex/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# DCRDEX uses its own auth
+      PROXY_AUTH_ADD: "false"  
+      APP_HOST: decred-dcrdex_web_1
+      APP_PORT: 5758
+  
+  web:
+    image: decred/dcrdex:v0.5.9
+    restart: on-failure
+    ports:
+      - 5758
```
Comment (nmfretz) [inline decred-dcrdex/docker-compose.yml:15]: Is it necessary to expose this port?

File: decred-dcrdex/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    volumes:
+      - ${APP_DATA_DIR}:/dex/.dexc
```
Comment (nmfretz) [inline decred-dcrdex/docker-compose.yml:17]: Are you wanting to mount the entire app directory or do you just want a subdirectory to store data from `/dex/.dexc` in.

you could consider mounting just a data directory like this:
`- ${APP_DATA_DIR}/data:/dex/.dexc`

**Review by nmfretz — COMMENTED** [review]
Very high quality PR. Fantastic work @peterzen! I have left some questions and suggested changes. I also took a look at the community app store implementation and everything looks good there.

File: decred-dcrdex/umbrel-app.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  Trade Bitcoin, Decred, Dogecoin, Dogecoin, ZCash and more peer-to-peer using atomic swaps.
+  
+  No trading fees. No KYC. You stay in control of your assets throughout the entire trading process.
+developer: The Decred Developers
+website: https://dex.decred.org
+dependencies: []
+repo: https://github.com/decred/dcrdex
+support: https://github.com/decred/dcrdex
+port: 5758
+icon: https://svgshare.com/i/pNb.svg
```
[the remark this answers is not in the document]
Comment (peterzen) [reply decred-dcrdex/umbrel-app.yml:17]: Moved the files into the Github, this should fix the issue.

File: decred-dcrdex/docker-compose.yml (hunk 1 above)
Comment (peterzen) [reply decred-dcrdex/docker-compose.yml:15]: No, this EXPOSE isn't necessary indeed.

**Comment by peterzen** [comment]
Thank you for the review @nmfretz, I've implemented your suggestions in the Community App Store repo which now works correctly.

**Pull request closed without merging**